README.md

```diff
@@ -568,6 +568,15 @@ extern "Rust" {
     /// program) the contents of a section of program memory, as bytes. Bytes
     /// written using this function will emerge from the interpreter's stderr.
     fn miri_write_to_stderr(bytes: &[u8]);
+
+    /// Miri-provided extern function to allocate memory from the interpreter.
+    /// 
+    /// This is useful when no fundamental way of allocating memory is
+    /// available, e.g. when using `no_std` + `alloc`.
+    fn miri_alloc(size: usize, align: usize) -> *mut u8;
+
+    /// Miri-provided extern function to deallocate memory.
+    fn miri_dealloc(ptr: *mut u8, size: usize, align: usize);
```

Comment:
Why not use the same signatures as `std::alloc::alloc` and `std::alloc::dealloc`?

Reply:
@RalfJung told me it would be easier to implement if we take `size` and `align` separately instead of one `Layout`. We are able to share most of the code with the `__rust_*` paths now.

Apart from that, there is no real user-facing reason for this approach. If the maintainers preferred using `Layout`, I'll look into it.

Reply:
For sure it's easier to implement this way. I don't have any particular preference, I'm just wary of these being a regrettable API inconsistency. They're a Rust interface, it would make sense to speak the Rust types for allocation.

I know I have had my own share of struggles writing code reads non-primitive types from an ABI boundary like this. We should really have a go-to resource for how to do that. So if you and @RalfJung want to move your project forward by going with this interface (for now?) instead of getting hung up on writing the required code, that wouldn't bother me.

Reply:
I view these as low-level machine interfaces; the nice Rust interfaces can be provided by a crate that wraps them.